.gitlab-ci.yml

```diff
@@ -7,12 +7,70 @@ variables:
   GIT_SUBMODULE_STRATEGY: recursive
   PROJECT_ALLOC_NAME: ${CI_PROJECT_NAME}_ci_${CI_PIPELINE_ID}
   BUILD_ROOT: ${CI_PROJECT_DIR}
+  ACCOUNT: -p pdebug -A radiuss
 
 # There are no tests for now
 stages:
   - allocate_resources
   - build
   - release_resources
+  - build_with_deps
+
+# Whether and how to update uberenv
+.run_update_uberenv: &run_update_uberenv |
+  [[ -n "${UPDATE_UBERENV}" ]] && ./scripts/gitlab/update-uberenv.sh "${UPDATE_UBERENV}"
```

Comment:
Am I understanding this right that you are now updating the uberenv.py script every PR?  Because if that is so that is not okay.... PR's should be testing the code that was changed by the developer and not injecting possible failures itself.

Reply:
`run_update_uberenv` is called only in the master workflow (only for master branch) _and_ if the `UPDATE_UBERENV` is set. In short: it is called _only_ if the master pipeline is triggered by uberenv (never in the development process).

Also the update is local to the pipeline, not commited. The goal is to trace if updating is safe. Next step could be to add a manual job to effectively update uberenv if it has be proved safe.

Reply:
Again, this manual job would appear only is such pipeline. Providing an easy way to update `uberenv`.

This may seem of little use in serac. It has already prove useful to me for uberenv validation. But another benefit is to demonstrate this cross-project pipeline mechanism that can be very useful if other cases.

Reply:
Why not have that pipeline under uberenv? Since you are verifying uberenv not serac?

Reply:
There is a part in Uberenv. Here, I’m simply allowing the master pipeline to be triggered with and updated version of Uberenv. Placing this in Uberenv would only duplicate the existing pipeline.

Not all in this PR is dedicated to testing uberenv:
- This PR introduces a master pipeline that will also install the dependencies. So that this install get checked from time to time. (Could be made _scheduled_ only, like once a week).
- With an additional 10 lines of script, I can allow uberenv to use it to test itself in serac context.

Reply:
Okay, I understand better now.  Though, having a pipeline updating code that is not apart of the branch it's testing is confusing. Unless it is very specifically named to be that way, like "master_with_updated_uberenv".  "master" is also a very overloaded word in this context...

Reply:
You may think of it as in uberenv were a submodule. This is just testing if this submodule can be udated safely in master.

I see your point though, and it would be good idea to make this pipeline appear as a specific one more clearly. I can’t do that in this PR though, it’s a suggestion I will post on Gitlab :).

Reply:
Note: The pipeline does not run the master workflow at the moment because this branch is a PR... It certainly adds to the confusion. Sorry about that.

A demo of the master workflow running is found [here](https://lc.llnl.gov/gitlab/bernede1/serac/pipelines/11805).
Also, a demo of serac pipeline when triggered by uberenv is [here](https://lc.llnl.gov/gitlab/bernede1/serac/pipelines/11810).

Reply:
In the latter we see that the pipeline is triggered by an upstream project. So it’s less confusing.

Reply:
So here's a dumb question: should uberenv just be a submodule?

Reply:
So if I understand correctly, this is a test for an updated copy of uberenv. It sounds like the choice is either duplicate the Serac pipelines in the uberenv project or add a test uberenv pipeline to all of the dependent projects. Is that right @adrienbernede ? I don't really have an opinion yet. I'm just trying to understand the options.